packages/react/fast-components-react-base/src/stack-panel/stack-panel.tsx

```diff
@@ -0,0 +1,580 @@
+import { isArray, isFunction } from "util";
+import { StackPanelClassNameContract } from "@microsoft/fast-components-class-name-contracts-base";
+import Foundation, { HandledProps } from "@microsoft/fast-components-foundation-react";
+import { classNames, Direction, RtlScrollConverter } from "@microsoft/fast-web-utilities";
+import { get, isNil, merge } from "lodash-es";
+import React from "react";
+import { toPx } from "@microsoft/fast-jss-utilities";
+import { Orientation } from "@microsoft/fast-web-utilities";
+import { DisplayNamePrefix, ResizeObserverClassDefinition } from "../utilities";
+import {
+    StackPanelHandledProps,
+    StackPanelProps,
+    StackPanelUnhandledProps,
+} from "./stack-panel.props";
+
+export interface StackPanelState {
+    isScrollable: boolean;
+    renderedRangeStartIndex: number;
+    renderedRangeEndIndex: number;
+    direction: Direction;
+}
+
+/**
+ * Used to store the pixel coordinates and span of items
+ * along the stackpanel's axis
+ */
+interface ItemPosition {
+    start: number;
+    span: number;
+    end: number;
+}
+
+class StackPanel extends Foundation<
+    StackPanelHandledProps,
+    StackPanelUnhandledProps,
+    StackPanelState
+> {
+    public static displayName: string = `${DisplayNamePrefix}StackPanel`;
+
+    public static defaultProps: Partial<StackPanelProps> = {
+        managedClasses: {},
+        virtualize: true,
+        neverVirtualizeIndexes: [],
+        itemSpan: 100,
+        preloadBufferLength: 1,
+        orientation: Orientation.vertical,
+        scrollLayoutUpdateDelay: 0,
+    };
+
+    private static DirectionAttributeName: string = "dir";
+
+    protected handledProps: HandledProps<StackPanelHandledProps> = {
+        itemSpan: void 0,
+        virtualize: void 0,
+        neverVirtualizeIndexes: void 0,
+        preloadBufferLength: void 0,
+        orientation: void 0,
+        managedClasses: void 0,
+        initiallyVisibleItemIndex: void 0,
+        onScrollChange: void 0,
+        scrollLayoutUpdateDelay: void 0,
+    };
+
+    // ref to the root element of the component
+    private rootElement: React.RefObject<HTMLDivElement> = React.createRef<
+        HTMLDivElement
+    >();
+
+    /**
+     * ref to the element which is the container all children are rendered into
+     * It is sized accomodate all children and is the container that is being scrolled
+     */
+    private itemContainerElement: React.RefObject<HTMLDivElement> = React.createRef<
+        HTMLDivElement
+    >();
+
+    private resizeDetector: ResizeObserverClassDefinition;
+    private itemPositions: ItemPosition[];
+    private viewportSpan: number = 0;
+    private itemContainerSpan: number = 0;
+    private maxScroll: number = 0;
+
+    /**
+     * Stores last scroll position from scroll events
+     */
+    private lastRecordedScroll: number = 0;
+
+    /**
+     * Stores last scroll position from scroll events
+     */
+    private scrollLayoutUpdateTimer: NodeJS.Timer;
```

Comment:
This type doesn't seem accurate to me - this won't always execute in a NodeJS environment

Reply:
used the window timer instead